examples/text_to_image/train_text_to_image_sdxl.py

```diff
@@ -1084,7 +1084,7 @@ def unwrap_model(model):
 
                 # Add noise to the model input according to the noise magnitude at each timestep
                 # (this is the forward diffusion process)
-                noisy_model_input = noise_scheduler.add_noise(model_input, noise, timesteps)
+                noisy_model_input = noise_scheduler.add_noise(model_input, noise, timesteps).to(dtype=weight_dtype)
```

Comment:
Let's do this conditionally if the dtype of `noisy_model_input` doesn't match with `vae.dtype`?

Reply:
Thanks for your feedback, actually after I tested several times, the dtype error actually happens in:
```
model_pred = unet(
                    noisy_model_input,
                    timesteps,
                    prompt_embeds,
                    added_cond_kwargs=unet_added_conditions,
                    return_dict=False,
                )[0]
```
So I think it may not affect the vae. Further, the model_pred will be `model_pred.float()` in the `loss` part.